pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -43,6 +43,29 @@
 
 
 class TestDataFrameSetItem:
+    @pytest.mark.parametrize(
+        argnames="arr", argvalues=[(4, 2), (4, 3), (4, 4), (4, 10), (4, 20), (4, 30)]
+    )
+    def test_setitem_size_incompatible_ndarray(self, arr):
+        data = DataFrame(np.zeros((4, 2)), columns=["A", "B"])
+        msg = (
+            "Dataframe column 'A' is being assigned to a 2D "
+            "array with more than two columns. Column assignment "
+            "accepts only 2D arrays with one column."
+        )
+        with pytest.raises(Exception, match=msg):
+            data["A"] = np.random.randn(arr[0], arr[1])
```

Comment:
Please add tests with loc and iloc

Reply:
I have done this but I'm unsure if I have put them in the right location.